prov/rxm/src/rxm.h

```diff
@@ -438,7 +438,10 @@ struct rxm_ep {
 	 * which segment length mast be calculated. */
 	uint32_t		sar_max_calc_seg_no;
 
-	struct rxm_buf_pool	buf_pools[RXM_BUF_POOL_MAX];
+	/* This is used only in non-FI_THREAD_SAFE case */
+	struct rxm_tx_buf	*inject_tx_buf;
+
+	struct rxm_buf_pool	*buf_pools;
```

Comment:
Is there a need to make this dynamically allocated since we're consuming an entry for inject buffer pool anyway?

Reply:
The individual buf pool should be dynamically allocated. I was only asking about the buf_pools array. Line 365 allocates the same size of memory as that of a statically allocated variable.

Reply:
Yes, right. Line 365 allocates the same size of memory, but `struct rxm_ep` structure has smaller size. As RxM widely uses `struct rxm_ep` structure, it would be great to keep it small as possible